core/worker.go

```diff
@@ -1174,7 +1174,7 @@ func (w *worker) prepareWork(genParams *generateParams, wo *types.WorkObject) (*
 	} else {
 		proposedWoHeader := types.NewWorkObjectHeader(newWo.Hash(), newWo.ParentHash(nodeCtx), newWo.Number(nodeCtx), newWo.Difficulty(), types.EmptyRootHash, newWo.Nonce(), newWo.Time(), newWo.Location())
 		proposedWoBody := types.NewWorkObjectBody(newWo.Header(), nil, nil, nil, nil, nil, nil, nodeCtx)
-		proposedWo := types.NewWorkObject(proposedWoHeader, proposedWoBody, nil, types.BlockObject)
+		proposedWo := types.NewWorkObject(proposedWoHeader, proposedWoBody, nil)
```

Comment:
why was this changed? i think we need this, because this can be used to build other types of work objects in the future. or is the idea to make a block work object by default and then convert it when it needs to be used ? i would like to understand more